test/tree_test.js

```diff
@@ -315,37 +295,47 @@ describe("Tree", () => {
       const cursor = tree.walk();
       const root = tree.rootNode;
 
-      // This should be the product node
       cursor.reset(root.children[0].children[0]);
-      expected = {
+      assertCursorState(cursor, {
         nodeType: 'product',
         nodeIsNamed: true,
         startPosition: {row: 0, column: 0},
         endPosition: {row: 0, column: 5},
         startIndex: 0,
         endIndex: 5
-      }
-      checkCursorNode(cursor, expected);
-      checkCursorNode(cursor.currentNode, expected);
+      });
+
+      cursor.gotoFirstChild()
+      assertCursorState(cursor, {
+        nodeType: 'variable',
+        nodeIsNamed: true,
+        startPosition: {row: 0, column: 0},
+        endPosition: {row: 0, column: 1},
+        startIndex: 0,
+        endIndex: 1
+      });
 
       cursor.reset(root);
       assert(!cursor.gotoParent());
     })
   });
 });
 
-function checkCursorNode(target, expected) {
-  if (target instanceof Parser.SyntaxNode) {
-    assert.equal(target.type, expected.nodeType);
-    assert.equal(target.isNamed, expected.nodeIsNamed);
-  } else {
-    assert.equal(target.nodeType, expected.nodeType);
-    assert.equal(target.nodeIsNamed, expected.nodeIsNamed);
-  }
-  assert.deepEqual(target.startPosition, expected.startPosition);
-  assert.deepEqual(target.endPosition, expected.endPosition);
-  assert.deepEqual(target.startIndex, expected.startIndex);
-  assert.deepEqual(target.endIndex, expected.endIndex);
+function assertCursorState(cursor, params) {
+  assert.equal(cursor.nodeType, params.nodeType);
+  assert.equal(cursor.nodeIsNamed, params.nodeIsNamed);
+  assert.deepEqual(cursor.startPosition, params.startPosition);
+  assert.deepEqual(cursor.endPosition, params.endPosition);
+  assert.deepEqual(cursor.startIndex, params.startIndex);
+  assert.deepEqual(cursor.endIndex, params.endIndex);
+
+  const node = cursor.currentNode
+  assert.equal(node.type, cursor.nodeType);
```

Comment:
I think the two options are equivalent because of lines 325 and 326, where we verify that the cursor matches the parameters. But maybe it's better to compare to `params.nodeType`, just for readability.